core/api/src/main/scala/org/virtuslab/ideprobe/protocol/ProjectRef.scala

```diff
@@ -1,10 +1,42 @@
 package org.virtuslab.ideprobe.protocol
 
+import com.typesafe.config.ConfigValueFactory
+import org.virtuslab.ideprobe.ConfigFormat
+import pureconfig.error.{CannotConvert, ConfigReaderFailures, ConvertFailure}
+import pureconfig.generic.semiauto.deriveReader
+import pureconfig.{ConfigCursor, ConfigReader, ConfigWriter}
+
 sealed trait ProjectRef
 
-object ProjectRef {
+object ProjectRef extends ConfigFormat {
   case object Default extends ProjectRef
   case class ByName(name: String) extends ProjectRef
 
   def apply(name: String): ProjectRef = ByName(name)
+
+  private val defaultProjectRefConfigReader: ConfigReader[ProjectRef.Default.type] = (cursor: ConfigCursor) => {
+    cursor.asString match {
+      case Left(failures) => Left[ConfigReaderFailures, ProjectRef.Default.type](failures)
+      case Right(refType) =>
+        if (refType == "default") Right[ConfigReaderFailures, ProjectRef.Default.type](Default)
+        else Left[ConfigReaderFailures, ProjectRef.Default.type](
+          ConfigReaderFailures(
+            ConvertFailure(
+              CannotConvert("", "ProjectRef.Default.type", "Cannot read data as default project"), cursor)
+          )
+        )
+    }
+  }
+
+  implicit val projectRefConfigReader: ConfigReader[ProjectRef] = {
+    possiblyAmbiguousAdtReader(
+      deriveReader[ByName],
+      defaultProjectRefConfigReader
+    )
+  }
+
+  implicit val projectRefConfigWriter: ConfigWriter[ProjectRef] = ConfigWriter.fromFunction {
```

Comment:
Writer for ByName should be derived. ConfigValueFactory only accepts basic types, it fails for your tuple.
```
  private val defaultProjectRefConfigReader: ConfigReader[ProjectRef.Default.type] =
    ConfigReader.fromStringOpt(s => if (s.toLowerCase == "default") Some(Default) else None)

  implicit val projectRefConfigReader: ConfigReader[ProjectRef] = {
    possiblyAmbiguousAdtReader(
      deriveReader[ByName],
      defaultProjectRefConfigReader
    )
  }
```